File: Samples/FilePicker/js/js/scenario2_MultiFile.js
//// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
//// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
//// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//// PARTICULAR PURPOSE.
////
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";
    var page = WinJS.UI.Pages.define("/html/scenario2_MultiFile.html", {
        ready: function (element, options) {
            element.querySelector(".open").addEventListener("click", pickMultipleFiles, false);
        }
    });

    function pickMultipleFiles() {
        // Clean scenario output
        WinJS.log && WinJS.log("", "sample", "status");

        // Create the picker object and set options
        var openPicker = new Windows.Storage.Pickers.FileOpenPicker();
        openPicker.viewMode = Windows.Storage.Pickers.PickerViewMode.list;
        openPicker.suggestedStartLocation = Windows.Storage.Pickers.PickerLocationId.documentsLibrary;
        openPicker.fileTypeFilter.replaceAll(["*"]);

        // Open the picker for the user to pick a file
        openPicker.pickMultipleFilesAsync().then(function (files) {
            if (files.size > 0) {
                // Application now has read/write access to the picked file(s)
                var outputString = "Picked files:\n";
                for (var i = 0; i < files.size; i++) {
                    outputString = outputString + files[i].name + "\n";
                }
                WinJS.log && WinJS.log(outputString, "sample", "status");
            } else {
                // The picker was dismissed with no selected file
                WinJS.log && WinJS.log("Operation cancelled.", "sample", "status");
            }
        });
    }
})();

File: Samples/FilePicker/cppwinrt/Scenario2_MultiFile.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario2_MultiFile.h"
#include <sstream>
#include "Scenario2_MultiFile.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Foundation::Collections;
using namespace Windows::Storage;
using namespace Windows::Storage::Pickers;
using namespace Windows::UI::Xaml;

namespace winrt::SDKTemplate::implementation
{
    Scenario2_MultiFile::Scenario2_MultiFile()
    {
        InitializeComponent();
    }

    fire_and_forget Scenario2_MultiFile::PickFilesButton_Click(IInspectable const&, RoutedEventArgs const&)
    {
        auto lifetime = get_strong();

        // Clear any previously returned files between iterations of this scenario
        OutputTextBlock().Text(L"");

        FileOpenPicker openPicker;
        openPicker.ViewMode(PickerViewMode::List);
        openPicker.SuggestedStartLocation(PickerLocationId::DocumentsLibrary);
        openPicker.FileTypeFilter().Append(L"*");
        IVectorView<StorageFile> files = co_await openPicker.PickMultipleFilesAsync();
        if (files.Size() > 0)
        {
            std::wstringstream output;
            output << L"Picked files:" << std::endl;
            // The StorageFile have read/write access to the picked files.
            // See the FileAccess sample for code that uses a StorageFile to read and write.
            for (StorageFile const& file : files)
            {
                output << std::wstring_view{ file.Name() } << std::endl;                
            }
            OutputTextBlock().Text(output.str());
        }
        else
        {
            OutputTextBlock().Text(L"Operation cancelled.");
        }
    }
}
